### ISSUE TITLE
Assist to convert between `Option::then` and if expressions

### ISSUE THREAD

**Issue by Veykril** [rust-analyzer/rust-analyzer#8413]
Now that `Option::then` is stable an assist to convert between it and if expressions would be quite nice to have I believe.
It should allow to convert between the two snippets here:
```rust
if cond {
	/* more expressions */
	Some(val)
} else {
	None
}
```
<->
```rust
cond.then(|| {
	/* more expressions */
	val
})
```
Note that `cond` might have to be wrapped in parentheses.

**Comment by flodiebold** [comment]
`then` -> `if` would actually have been pretty much covered by `inline method`, except that we don't simplify immediately-evaluated closures...

**Comment by Veykril** [comment]
That is true, but the assist moves the `Some` wrapping into expressions, inlining it would wrap the entire closure body in a `Some`.
